### PR TITLE
Fixes training/validation overlap #143, #233, #253, and #259

### PR DESCRIPTION
**Pull request by yalaudah**
This PR fixes the following issues: #143, #233, #253, and partially solves issue #259.

This solution is not ideal, it limits training and validation to one direction (inline or crossline). If the training results are significantly affected, we can add code to train and val in both directions. 

Remaining: 
- [ ] Add unit tests
- [ ] Test model performance with and without these changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,12 +142,12 @@ To prepare the data for the experiments (e.g. split into train/val/test), please
...
 
 # For section-based experiments
-python prepare_dutchf3.py split_train_val section --data_dir=${data_dir} --label_file=train/train_labels.npy --output_dir=splits
+python prepare_dutchf3.py split_train_val section --data_dir=${data_dir} --label_file=train/train_labels.npy --output_dir=splits  --split_direction=crossline
 
 
 # For patch-based experiments
 python prepare_dutchf3.py split_train_val patch --data_dir=${data_dir} --label_file=train/train_labels.npy --output_dir=splits \
---stride=50 --patch_size=100
+--stride=50 --patch_size=100 --split_direction=crossline
```
Comment (maxkazmsft) [inline README.md:150]: We're splitting in both directions, right? So this flag is no longer needed?

File: contrib/experiments/interpretation/penobscot/local/test.py [hunk 2]
```diff
@@ -268,9 +268,9 @@ def _tensor_to_numpy(pred_tensor):
 
     logger.info("Starting training")
     if debug:
-        evaluator.run(test_loader, max_epochs=1, epoch_length = 1)
+        evaluator.run(test_loader, max_epochs=1, epoch_length=1)
```
Comment (maxkazmsft) [inline contrib/experiments/interpretation/penobscot/local/test.py:271]: Shouldn't this be config.TRAIN.BATCH_SIZE_PER_GPU for epoch_length?

File: contrib/scripts/ablation.sh [hunk 3]
```diff
@@ -3,22 +3,22 @@
 source activate seismic-interpretation
 
 # Patch_Size 100: Patch vs Section Depth
-python scripts/prepare_dutchf3.py split_train_val patch --data_dir=/mnt/dutch --stride=50 --patch_size=100
+python scripts/prepare_dutchf3.py split_train_val patch --data_dir=/mnt/dutch --stride=50 --patch_size=100 --split_direction=crossline
```
Comment (maxkazmsft) [inline contrib/scripts/ablation.sh:6]: Again, this flag is not needed, right? Same for the other lines below.

File: docker/Dockerfile [hunk 4]
```diff
@@ -48,7 +48,7 @@ RUN data_dir="/home/username/data/dutch" && \
     cd scripts && \
     source activate seismic-interpretation && \
     python prepare_dutchf3.py split_train_val section --data-dir=${data_dir}/data --label_file=train/train_labels.npy --output_dir=splits && \
-    python prepare_dutchf3.py split_train_val patch --data-dir=${data_dir}/data --label_file=train/train_labels.npy --output_dir=splits --stride=50 --patch_size=100 && \
+    python prepare_dutchf3.py split_train_val patch --data-dir=${data_dir}/data --label_file=train/train_labels.npy --output_dir=splits --stride=50 --patch_size=100 --split_direction=crossline && \
```
Comment (maxkazmsft) [inline docker/Dockerfile:51]: same comment about splits

File: scripts/prepare_dutchf3.py [hunk 5]
```diff
@@ -332,79 +292,102 @@ def split_alaudah_et_al_19(data_dir, stride, patch_size, fraction_validation=0.2
...
+            per_val=per_val,
+            log_config=log_config,
+        )
 
 
 if __name__ == "__main__":
     """Example:
     python prepare_data.py split_train_val section --data_dir=data \
-        --label_file=label_file.npy --output_dir=splits --slice_steps=2
+        --label_file=label_file.npy --output_dir=splits --split_direction=crossline --section_stride=2
```
Comment (maxkazmsft) [inline scripts/prepare_dutchf3.py:382]: same comment about split direction

File: README.md (hunk 1 above)
Comment (yalaudah) [reply README.md]: we discussed this offline and agreed to add a "both" option to the split_direction

File: contrib/scripts/ablation.sh (hunk 3 above)
Comment (yalaudah) [reply contrib/scripts/ablation.sh]: we discussed this offline and agreed to add a "both" option to the split_direction

File: contrib/experiments/interpretation/penobscot/local/test.py (hunk 2 above)
Comment (yalaudah) [reply contrib/experiments/interpretation/penobscot/local/test.py:271]: This seems to be the same as the latest code from the correctness branch (after the ignite upgrade). Here's the code:

https://github.com/microsoft/seismic-deeplearning/blob/cf336ee059612e621550475e34bc943b0498dbaa/contrib/experiments/interpretation/penobscot/local/test.py#L286-L288

Comment (maxkazmsft) [reply contrib/experiments/interpretation/penobscot/local/test.py:271]: OK then let's just leave it as is

**Pull request merged**